File: 0101-symmetric-tree/0101-symmetric-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean isSymmetric(TreeNode root) {
        
        return sym(root,root);
        
    }
    boolean sym(TreeNode A, TreeNode B){
        if(A == null && B == null) return true;
        if(A == null || B == null) return false;

        if(A.val != B.val){
            return false;
        }

        return sym(A.left, B.right) && sym(A.right , B.left);
    }
}

File: 0101-symmetric-tree/0101-symmetric-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSymmetric(self, root: Optional[TreeNode]) -> bool:
        if not root:
            return True

        def helper(left , right):
            if not left and not right:
                return True
            if not left or not right:
                return False
            # if left.val  != right.val:
            #     return False

            # leftval = left.left.val
            # rightval = right.right.val

            # if leftval != rightval:
            #     return False
            
            return left.val == right.val and  helper(left.left , right.right) and helper(left.right , right.left)

        return helper(root.left , root.right)
    